internal/services/eventhub/eventhub_namespace_customer_managed_key_resource.go

```diff
@@ -208,8 +237,31 @@ func expandEventHubNamespaceKeyVaultKeyIds(input []interface{}) (*[]namespaces.K
 	return &results, nil
 }
 
-func flattenEventHubNamespaceKeyVaultKeyIds(input *namespaces.Encryption) ([]interface{}, error) {
-	results := make([]interface{}, 0)
+// Get the user managed id for the custom key if one exists
+func getUserManagedIdentity(input *namespaces.Encryption) *[]interface{} {
+	if input == nil || input.KeyVaultProperties == nil {
+		return nil
+	}
+
+	// we can only have a single managed id, so just take the first one and call it a day
+	for _, item := range *input.KeyVaultProperties {
+		if item.Identity != nil {
+			return &[]interface{}{
+				map[string]interface{}{
+					"type":         "UserAssigned",
+					"identity_ids": []string{*item.Identity.UserAssignedIdentity},
+					"principal_id": "",
+					"tenant_id":    "",
+				},
+			}
+		}
+	}
```

Comment:
this seems odd to me, it looks like each key vault can have a different managed identity?

Reply:
Strangely, that is not the case. While multiple keys in multiple vaults can be assigned, only one user assigned ID can be assigned to the event hub encryption resource. They can only be added one at a time, and adding a new one removes the old one.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/17836372/8b8ddc83-8f2a-42c4-9345-9db01b9b6e3e)